### ISSUE TITLE
Configuring ksql to insert json payload with a string key for cassandra source connector.

### ISSUE THREAD

**Issue by sjmittal** [lensesio/stream-reactor#642]
Hi,
I am using kafka-connect-cassandra-1.2.3-2.1.0 source connector to move data from cassandra to kafka topic. I am able to set this up but I am having hard time sending the json payload with a string key to kafka.

My ksql is something like:`
INSERT INTO topic SELECT key, capture_time, transaction_name FROM table PK capture_time WITHFORMAT JSON INCREMENTALMODE=TIMESTAMP`

However in kafka topic it inserts something like:
key: null
value: 
```
{
  "schema": {
    ...
  },
  "payload": {
    "key": "agent1",
    "capture_time": 1579087876000,
    "transaction_name": "transaction 1"
  }
}
```
However I want is:
key: `agent1`
value: 
```
{
 "capture_time": 1579087876000,
 "transaction_name": "transaction 1"
}
```

I have tinkered with vaious options of ksql like WITHTYPE , WITHUNWRAP but nothing has come to help. Also docs on https://docs.lenses.io/sql/connectSql.htm do not provide much insight into how to do nor the cassandra connector page. Is there any way I can achieve what I want.

**Comment by andrewstevenson** [comment]
@sjmittal Each connector implements KCQL options differently, WITHFORMAT, WITHUNWARP are not support on this connector. If you want to write the records from as Json into Kafka, which I advise against (use avro) you should set the converters on the Connect cluster workers to use the Json serializer. See https://debezium.io/documentation/reference/1.0/configuration/avro.html

**Comment by sjmittal** [comment]
@andrewstevenson for now I only have option to write as json to kafka topic, since other part of the pipeline is done and assumes this format.
From the docs it is not clear as what all options of KCQL and options for cassandra connect config are available to use or supported.
Like you say that WITHUNWARP is not supported but when I check this blog post, here its mentioned how this options is supported.
ref: https://lenses.io/blog/2018/03/cassandra-to-kafka/
This blog post also mentions on how we can configure the cluster for various use cases using cassandra connect options.

However doc: https://docs.lenses.io/connectors/source/cassandra.html is not all that detailed.

Anyway what I have understood is that I need to use convertors but it is still not clear as how and where to use that. As explained above the record in topic is saved as Json format, but it has all this extra information which I want to remove and also I need to specify a (string) key for that record. 
How can I do it?

Like using KCQL I can somehow of specify which column maps as (string) key and only send payload (without schema) as Json value. Is this possible?

**Comment by sjmittal** [comment]
OK So I have manage to implement the WITHKEY and WITHFORMAT JSON options to the cassandra source connector.

Now a kcql like this can be processed by the cassandra source connector.

`INSERT INTO topic SELECT id, time, name FROM rollup_0 PK time WITHFORMAT JSON WITHUNWRAP INCREMENTALMODE=TIMESTAMP WITHKEY(key)`

Following code had to be added to `processRow `method of `CassandraTableReader.scala`.
So this just helps me solve for my case. However we should add more comprehensive options to cassandra connector especially able to send messages with different formats and different keys.

```
if (config.isUnwrapping) {
      if (config.getFormatType == FormatType.JSON) {
        val keys = config.getWithKeys
        val v = structColDefs.map(d => d.getName -> row.getObject(d.getName)).toMap
        val structValue = Json(DefaultFormats).write(v)
        if(keys.isEmpty) {
          new SourceRecord(sourcePartition, Map(primaryKeyCol -> offset), topic, Schema.STRING_SCHEMA, structValue)
        } else {
          val keyValue = keys.map(k => row.getObject(k)).mkString(",")
          new SourceRecord(sourcePartition, Map(primaryKeyCol -> offset), topic, Schema.STRING_SCHEMA, keyValue, Schema.STRING_SCHEMA, structValue)
        }
      } else {
        val structValue = structColDefs.map(d => d.getName).map(name => row.getObject(name)).mkString(",")
        new SourceRecord(sourcePartition, Map(primaryKeyCol -> offset), topic, Schema.STRING_SCHEMA, structValue)
      }
    }
```

instead of existing code of just
```
if (config.isUnwrapping) {
      val structValue = structColDefs.map(d => d.getName).map(name => row.getObject(name)).mkString(",")
      new SourceRecord(sourcePartition, Map(primaryKeyCol -> offset), topic, Schema.STRING_SCHEMA, structValue)
}
```

**Comment by andrewstevenson** [comment]
@sjmittal Nice, can you submit a PR?

**Comment by sjmittal** [comment]
I have created a PR https://github.com/lensesio/stream-reactor/pull/646
See if it makes sense.

**Comment by andrewstevenson** [comment]
@sjmittal Added a comment.

**Comment by andrewstevenson** [comment]
Resolved in #646
